www/js/controller.js

```diff
@@ -807,20 +828,42 @@ define([
 		/* Stub to update connections in graph */
 		addQuantity: function(source, destinations){
 		},
+		handleVariableType: function(event){
+			// Summary : Sets variableType to Unknown/Parameter/Dynamic
+			// Value is not allowed when variableType is Unknown
+			// Value is handled when variableType is parameter or dynamic.
+			console.log("********************* in handleVariableType");
+			var _variableType = event.target.value;
+			if(_variableType === this._model.active.getVariableType(this.currentID)){
+				return;
+			}
+			
+			this.variableTypeControls(this.currentID, _variableType);
+			if(this._mode !== "AUTHOR"){
+				this.applyDirectives(this._PM.processAnswer(this.currentID, 'variableType', _variableType));
```

Comment:
This was the whole idea of constructor - con-author and con-student architecture. If you have to do this then it would be better to have a function in con-student and an empty function with the same name in constructor. So what would happen it will be called in student and then it wont be called in author mode. and then the idea of setting a field to be disabled is the job of apply directives. Please would it be possible to do these things there only.